umh-core/pkg/fsm/dataflowcomponent/dfc_test.go

```diff
@@ -0,0 +1,277 @@
+		componentConfig    dataflowcomponent.DataFlowComponentConfig
+		tempConfigFilePath string
+	)
+
+	BeforeEach(func() {
+		ctx = context.Background()
+		mockConfigManager = NewMockBenthosConfigManager()
+
+		// Create a temporary directory for config files
+		tempDir, err := os.MkdirTemp("", "dataFlowComponent-test")
+		Expect(err).NotTo(HaveOccurred())
+		tempConfigFilePath = filepath.Join(tempDir, "test-config.yaml")
+
+		// Basic component config
+		componentConfig = dataflowcomponent.DataFlowComponentConfig{
+			Name:         "test-component",
+			DesiredState: "stopped",
+			ServiceConfig: benthosserviceconfig.BenthosServiceConfig{
+				Input: map[string]interface{}{
+					"generate": map[string]interface{}{
+						"mapping":  "root = \"hello world from test!\"",
+						"interval": "1s",
+						"count":    0,
+					},
+				},
+				Output: map[string]interface{}{
+					"stdout": map[string]interface{}{},
+				},
+			},
+		}
+
+		// Create a new DataFlowComponent
+		testComponent = dataflowcomponent.NewDataFlowComponent(componentConfig, mockConfigManager)
+		Expect(testComponent).NotTo(BeNil())
+
+		// Complete the lifecycle creation process (to_be_created -> creating -> created -> stopped)
+		// First reconcile to add to config and transition from to_be_created to creating
+		err, _ = testComponent.Reconcile(ctx, 1)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Second reconcile to complete creation (creating -> created)
+		err, _ = testComponent.Reconcile(ctx, 2)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Now the FSM should be in the stopped operational state as configured
+		Expect(testComponent.GetCurrentFSMState()).To(Equal(dataflowcomponent.OperationalStateStopped))
+
+		// Reset mock flags after initialization
+		mockConfigManager.addCalled = false
+		mockConfigManager.removeCalled = false
+		mockConfigManager.updateCalled = false
+		mockConfigManager.checkExistenceCalled = false
+	})
+
+	AfterEach(func() {
+		// Clean up the temporary directory
+		if tempConfigFilePath != "" {
+			os.RemoveAll(filepath.Dir(tempConfigFilePath))
+		}
+	})
+
+	Describe("State Transitions", func() {
+		It("should start in the Stopped state", func() {
+			Expect(testComponent.GetCurrentFSMState()).To(Equal(dataflowcomponent.OperationalStateStopped))
+		})
+
+		It("should transition to Active when the desired state is set to Active", func() {
+			// Set the desired state to Active
+			err := testComponent.SetDesiredFSMState(dataflowcomponent.OperationalStateActive)
+			Expect(err).NotTo(HaveOccurred())
+
+			// First reconcile call should transition from Stopped to Starting
+			err, reconciled := testComponent.Reconcile(ctx, 3)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(reconciled).To(BeTrue())
+			Expect(testComponent.GetCurrentFSMState()).To(Equal(dataflowcomponent.OperationalStateStarting))
+
+			// Second reconcile call should transition from Starting to Active
+			err, reconciled = testComponent.Reconcile(ctx, 4)
```

Comment:
also, how is this working without configuring any underlying mocks that simulate the behaviour of benthos?

Reply:
checkout the mock adapter